lib/db_connection/connection.ex

```diff
@@ -563,10 +574,23 @@ defmodule DBConnection.Connection do
     end
   end
 
+  defp handle_timeout(%{client: nil, idle_timeout: idle_timeout,
+                        idle_hibernate: true,
+                        hibernate_timer: hibernate_timer} = s) do
+    cancel_timer(hibernate_timer)
+    hibernate_timer = Process.send_after(self(), :timeout, idle_timeout)
+    # hibernate_timer = start_timer(self(), idle_timeout, :trigger_hibernate)
```

Comment:
When the timer arrives we need to call the logic that is currently done by `handle_info(:timeout, ...)` because the timer is replacing that timeout. The reason it fails is because we don't call `apply(mod, :ping, [state])` anymore.